src/world/Objects/Units/Creatures/Creature.cpp

```diff
@@ -884,14 +884,14 @@ void Creature::SaveToDB()
         m_spawn->channel_spell = 0;
         m_spawn->MountedDisplayID = getMountDisplayId();
 
-#if VERSION_STRING < WotLK
-        m_spawn->Item1SlotEntry = getVirtualItemEntry(MELEE);
-        m_spawn->Item2SlotEntry = getVirtualItemEntry(OFFHAND);
-        m_spawn->Item3SlotEntry = getVirtualItemEntry(RANGED);
+#if VERSION_STRING >= WotLK
+        m_spawn->itemEquipSlots[MELEE] = getVirtualItemSlotId(MELEE);
+        m_spawn->itemEquipSlots[OFFHAND] = getVirtualItemSlotId(OFFHAND);
+        m_spawn->itemEquipSlots[RANGED] = getVirtualItemSlotId(RANGED);
 #else
-        m_spawn->Item1SlotEntry = getVirtualItemSlotId(MELEE);
-        m_spawn->Item2SlotEntry = getVirtualItemSlotId(OFFHAND);
-        m_spawn->Item3SlotEntry = getVirtualItemSlotId(RANGED);
+        m_spawn->itemEquipSlots[MELEE] = getVirtualItemDisplayId(MELEE);
+        m_spawn->itemEquipSlots[OFFHAND] = getVirtualItemDisplayId(OFFHAND);
+        m_spawn->itemEquipSlots[RANGED] = getVirtualItemDisplayId(RANGED);
```

Comment:
Here for Classic and TBC instead of 'getVirtualItemDisplayId' use 'getVirtualItemEntry'. If you are working in Wotlk you don't see the function but it's there 🙂

Reply:
its handled with precompiler macro already + I splitted these functions, rather than have several areas handled with macros, i thought better to handle entire functions instead